### PR TITLE
WIP: feat(resolver): update article resolvers due to draft and article restructure

### PR DESCRIPTION
**Pull request by devformatters**
- Replace article root object with latest linked draft
- Revise all article sub field resolvers
- Revise `node` and `nodes` resolver for article

### CODE REVIEW

File: src/connectors/draftService.ts [hunk 1]
```diff
@@ -56,4 +56,10 @@ export class DraftService extends BaseService {
       .from(this.table)
       .where({ authorId })
       .andWhereNot({ publishState: PUBLISH_STATE.published })
+
+  /**
+   * Find draft by media hash.
+   */
+  findByMediaHash = async (mediaHash: string) =>
```
Comment (robertu7) [inline src/connectors/draftService.ts:63]: looks we need to add indexes for draft fields

File: src/queries/system/nodes.ts [hunk 2]
```diff
@@ -69,7 +69,16 @@ const resolver: QueryToNodesResolver = async (
...
   // fetch by bulk
-  const articles = await articleService.dataloader.loadMany(articleDbIds)
+  const articles = await Promise.all(
+    (await articleService.dataloader.loadMany(articleDbIds))
+      .map(async (node: Item | Error) => {
+        if (!node || node instanceof Error) {
+          return node
+        }
+        // fetch data from latest linked draft
+        return draftService.dataloader.load(node.draftId)
```
Comment (robertu7) [inline src/queries/system/nodes.ts:79]: Would it be better if use`.loadMany` here too?

**Review by robertu7 — APPROVED** [review]
🔢

**Review by guoliu — APPROVED** [review]
Looks good 🌽

**Comment by devformatters** [comment]
Drop it cuz there's a new PR for this.

**Pull request closed without merging**